src/QsCompiler/SyntaxProcessor/StatementVerification.fs

```diff
@@ -250,15 +315,21 @@ let NewConditionalBlock comments location context (qsExpr : QsExpression) =
     let block body = condition, QsPositionedBlock.New comments (Value location) body
     new BlockStatement<_>(block), Array.concat [errs; autoGenErrs]
 
-/// Given a conditional block for the if-clause of a Q# if-statement, a sequence of conditional blocks for the elif-clauses, 
+/// Given a conditional block for the if-clause of a Q# if-statement, a sequence of conditional blocks for the elif-clauses,
 /// as well as optionally a positioned block of Q# statements and its location for the else-clause, builds and returns the complete if-statement.  
 /// Throws an ArgumentException if the given if-block contains no location information. 
-let NewIfStatement (ifBlock : TypedExpression * QsPositionedBlock, elifBlocks, elseBlock : QsNullable<QsPositionedBlock>) = 
-    let location = (snd ifBlock).Location |> function 
-        | Null -> ArgumentException "no location is set for the given if-block" |> raise
-        | Value loc -> loc 
-    let condBlocks = seq { yield ifBlock; yield! elifBlocks; }
-    QsConditionalStatement.New (condBlocks, elseBlock) |> QsConditionalStatement |> asStatement QsComments.Empty location LocalDeclarations.Empty
+let NewIfStatement context (ifBlock : struct (TypedExpression * QsPositionedBlock)) elifBlocks elseBlock =
```

Comment:
If it is not a struct tuple, F# flattens the tuple contents into the argument list, creating a method visible from C# as `NewIfStatement(context, ifCond, ifBlock, elifBlocks, elseBlock)`. Maybe that's why the original code had all of the arguments wrapped in a tuple (F# doesn't flatten nested tuples), but [F# guidelines recommend using struct tuples for C# interop](https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/tuples#interoperation-with-c-tuples).

Reply:
The conversion on both sides in this case seems silly though, and all other methods here behave differently. Since we have a lot of F# data types, using structs all the way through would be a major inconvenience.

Reply:
Based on our discussion, I replaced the struct tuple with a normal tuple, and unpacked the the `ifBlock` tuple when calling it from C#.